fatiando/fatiando
# Enable duecredit for magdir

Fixes #222 and test for general rollout of #292

Adds the ability to easily get the citation of a function that was used.

To test:
1. Install duecredit: `pip install duecredit`

(This should be doable without installing it globally, but that is not working
for me. I may be missing something here with importing the module, because
this is what `our_duecredit.py` is supposed to enable, if I understand their
docs correctly.)

Then, either:
1. Call the script with `python -m duecredit
gravmag_magdir_dipolemagdir.py`.

OR

1. Set an environment variable: `DUECREDIT_ENABLE=yes`.
2. Run `python cookbook\gravmag_magdir_dipolemagdir.py`.

Any output from the script will be shown first, followed by something like:

    DueCredit Report:
    - Estimates total mag. dir. of approx. spherical bodies /
    fatiando.gravmag.magdir:DipoleMagDir (v 0.4-21-gcb9ce97-dirty) [1]
    - Scientific tools library / numpy (v 1.10.4) [2]

    1 package cited
    0 modules cited
    1 function cited

    References
    ----------

    [1] Oliveira, V.C. et al., 2015. Estimation of the total magnetization direction
    of approximately spherical bodies. Nonlin. Processes Geophys., 22(2),
    pp.215-232.

### Checklist:

- [x] Make tests for new code (at least 80% coverage)
- [x] Create/update docstrings
- [x] Include relevant equations and citations in docstrings
- [x] Docstrings follow the style conventions
- [x] Code follows PEP8 style conventions
- [x] Code and docs have been spellchecked
- [x] Include new dependencies in `doc/install.rst`, `requirements.txt`, `environment.yml`, `ci/requirements-conda.txt` and `ci/requirements-pip.txt`.
- [x] Documentation builds properly (run `cd doc; make` locally)
- [ ] Changelog entry (leave for last to avoid conflicts)
